Don't force the use of clang. Instead, use the compiler MacPorts specified.
--- build.sh.orig	2020-06-08 18:11:53.000000000 -0500
+++ build.sh	2020-06-14 17:26:47.000000000 -0500
@@ -34,7 +34,6 @@
 	plat_files="$plat_files ../src/compat/dummy.c"
 	plat_files="$plat_files ../src/compat/clearenv.c "
 	plat_cflags="$plat_cflags -include ../src/compat/macosx.h"
-	CC=clang
 	;;
 	FreeBSD)
 	plat_files="$plat_files ../src/compat/dummy.c"
